pkg/registry/flowcontrol/ensurer/prioritylevelconfiguration.go

```diff
@@ -18,192 +18,144 @@ package ensurer
 
 import (
 	"context"
-	"errors"
 	"fmt"
 
 	flowcontrolv1beta3 "k8s.io/api/flowcontrol/v1beta3"
 	"k8s.io/apimachinery/pkg/api/equality"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
-	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/util/sets"
 	flowcontrolclient "k8s.io/client-go/kubernetes/typed/flowcontrol/v1beta3"
 	flowcontrollisters "k8s.io/client-go/listers/flowcontrol/v1beta3"
 	flowcontrolapisv1beta3 "k8s.io/kubernetes/pkg/apis/flowcontrol/v1beta3"
 )
 
-var (
-	errObjectNotPriorityLevel = errors.New("object is not a PriorityLevelConfiguration type")
-)
+// WrapBootstrapPriorityLevelConfigurations creates a generic representation of the given bootstrap objects bound with their operations
+func WrapBootstrapPriorityLevelConfigurations(client flowcontrolclient.PriorityLevelConfigurationInterface, lister flowcontrollisters.PriorityLevelConfigurationLister, boots []*flowcontrolv1beta3.PriorityLevelConfiguration) BootstrapObjects {
+	return &bootstrapPriorityLevelConfigurations{
+		priorityLevelConfigurationClient: priorityLevelConfigurationClient{
+			client: client,
+			lister: lister},
+		boots: boots,
+	}
+}
 
-// PriorityLevelEnsurer ensures the specified bootstrap configuration objects
-type PriorityLevelEnsurer interface {
-	Ensure([]*flowcontrolv1beta3.PriorityLevelConfiguration) error
+type priorityLevelConfigurationClient struct {
+	client flowcontrolclient.PriorityLevelConfigurationInterface
+	lister flowcontrollisters.PriorityLevelConfigurationLister
 }
 
-// PriorityLevelRemover is the interface that wraps the
-// RemoveAutoUpdateEnabledObjects method.
-//
-// RemoveAutoUpdateEnabledObjects removes a set of bootstrap
-// PriorityLevelConfiguration objects specified via their names.
-// The function removes an object only if automatic update
-// of the spec is enabled for it.
-type PriorityLevelRemover interface {
-	RemoveAutoUpdateEnabledObjects([]string) error
+type bootstrapPriorityLevelConfigurations struct {
+	priorityLevelConfigurationClient
```

Comment:
nit: can we make it a named field:

Reply:
We could but what would be the benefit? The costs would include writing more methods because of promotions that are no longer there.